### ISSUE TITLE
"Symbol cannot be converted to a string" in executing lodash.js

### ISSUE THREAD

**Issue by jonathanbp** [paulbartrum/jurassic#155]
We are loading [lodash.js](https://lodash.com) in a Jurassic v3.0.0 ScriptEngine using `Engine.Execute`:

```cs
Engine.Execute(Env.ReadTextRessource("lodash.js"));
```

which worked in v2.2.2 but fails in v3.0.0 with:

```
Jurassic.JavaScriptException: TypeError: Cannot convert a Symbol value to a string.
   at Jurassic.TypeConverter.ToString(Object value)
   at baseGetTag(ScriptEngine , Scope , Object , FunctionInstance , Object[] )
   at Jurassic.Library.UserDefinedFunction.CallLateBound(Object thisObject, Object[] argumentValues)
   at Jurassic.Library.FunctionInstance.CallWithStackTrace(String path, String function, Int32 line, Object thisObject, Object[] argumentValues)
   at isFunction(ScriptEngine , Scope , Object , FunctionInstance , Object[] )
   at Jurassic.Library.UserDefinedFunction.CallLateBound(Object thisObject, Object[] argumentValues)
   at Jurassic.Library.FunctionInstance.CallWithStackTrace(String path, String function, Int32 line, Object thisObject, Object[] argumentValues)
   at baseIsNative(ScriptEngine , Scope , Object , FunctionInstance , Object[] )
   at Jurassic.Library.UserDefinedFunction.CallLateBound(Object thisObject, Object[] argumentValues)
   at Jurassic.Library.FunctionInstance.CallWithStackTrace(String path, String function, Int32 line, Object thisObject, Object[] argumentValues)
   at getNative(ScriptEngine , Scope , Object , FunctionInstance , Object[] )
   at Jurassic.Library.UserDefinedFunction.CallLateBound(Object thisObject, Object[] argumentValues)
   at Jurassic.Library.FunctionInstance.CallWithStackTrace(String path, String function, Int32 line, Object thisObject, Object[] argumentValues)
   at runInContext(ScriptEngine , Scope , Object , FunctionInstance , Object[] )
   at Jurassic.Library.UserDefinedFunction.CallLateBound(Object thisObject, Object[] argumentValues)
   at Jurassic.Library.FunctionInstance.CallWithStackTrace(String path, String function, Int32 line, Object thisObject, Object[] argumentValues)
   at anonymous(ScriptEngine , Scope , Object , FunctionInstance , Object[] )
   at Jurassic.Library.UserDefinedFunction.CallLateBound(Object thisObject, Object[] argumentValues)
   at Jurassic.Library.FunctionInstance.Call(Object thisObj, Object[] arguments)
   at Jurassic.Library.FunctionInstance.__STUB__Call(ScriptEngine engine, Object thisObj, Object[] args)
   at Jurassic.Library.ClrStubFunction.CallLateBound(Object thisObject, Object[] argumentValues)
   at Jurassic.Library.FunctionInstance.CallWithStackTrace(String path, String function, Int32 line, Object thisObject, Object[] argumentValues)
   at global_(ScriptEngine , Scope , Object )
   at Jurassic.Compiler.GlobalMethodGenerator.Execute(ScriptEngine engine)
   at Jurassic.CompiledScript.Execute(ScriptEngine engine)
   at Jurassic.ScriptEngine.Execute(ScriptSource source)
   at Jurassic.ScriptEngine.Execute(String code)
```

I haven't been able to pinpoint where in lodash.js the error originates exactly or whether lodash.js does something not according to standards, but I guess the error is relevant either way?

**Comment by kpreisser** [comment]
I think this is because Jurassic 3.0.0 has been released from a recent commit of the `master` branch where some ES2015 types like `Symbol` and `Promise` are available from the global object, but are only partially implemented (meaning they don't support every aspect of the specification that is supported by ES2015 runtimes).

Then, when a library checks e.g. if `typeof Symbol` is available, it will find and use it instead of a (fully working) polyfill.

Maybe we need to remove those types from the global object until they are fully implemented. (For the same reason, in my fork I removed the `Promise` object from the global object, so that I can use a polyfill.)

**Comment by jonathanbp** [comment]
You're right. I added a `Engine.SetGlobalValue("Symbol", null)` and the loads.js execution worked. I think either doing some documentation or removing the types is probably a good idea. 

On another note, I tried setting the compatibility-mode to `Jurassic.CompatibilityMode.ECMAScript3` but that did not work. I would have expected that to not include `Symbol`.

**Comment by paulbartrum** [comment]
Is this working now after commit 0be8e0bd23d61a2b69c7cae1ca1fc34aba8d9c15?

**Comment by paulbartrum** [comment]
Fixed by 72b6d28b16505edcabc5248ca63996fcc3779118
